R/apply_theme.R

```diff
@@ -90,14 +96,36 @@ define_theme <- function(strata = NULL,
     
   }
   
-  if (!base::is.logical(grid)) {
+  if (base::is.logical(grid)) {
     
-    base::warning("Invalid argument for `grid`. Please use a boolean to indicate whether you want a background grid. Setting to default `FALSE`.")
-    theme[["grid"]] <- FALSE
+    if (grid == TRUE) {
+      
+      theme[["grid"]] <- list("major" = TRUE,
+                              "minor" = FALSE)
+      
+    } else {
+      
+      theme[["grid"]] <- grid 
+      
+    }
+    
+  } else if (is.list(grid)) {
+    
+    if (("major" %in% names(grid)) | ("minor" %in% names(grid))) {
+      
+      theme[["grid"]] <- grid
+      
+    } else {
+      
+      base::warning("Invalid argument for `grid`. Please use a boolean or a list to indicate whether you want a background grid. Setting to default `FALSE`.")
```

Comment:
Also here. I'm not a fan of giving warnings for every argument This is really annoying as a user because you did not do anything wrong.

Reply:
Default behaviour should be described in roxygen2 documentation

Reply:
Well, that's why the warnings are only triggered if the user did anything wrong :D If you don't manually specify it, it's `FALSE` and doesn't trigger a warning. If you manually set it to `TRUE` or `FALSE` you don't get a warning. If you set it to a list that has at least one of ["major", "minor"] as members, you don't get a warning. Only if none of these cases are true, a warning is given.

Reply:
I prefer errors if a user misspecifies an argument but we can discuss after CRAN release.

Reply:
Good discussions. Let's agree on a common theme. I agree that that there should be _some_ action triggered when user does something wrong and alternative behaviour is applied. Otherwise it's a nightmare to debug. 
The question is if misspecifying parameters should actually cause the function to stop or merely warn. ggplot2 often just warns, depending on the severity of the issue.